pkg/reconciler/revision/resources/deploy.go

```diff
@@ -169,8 +181,23 @@ func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error)
 
 	// If concurrencyStateEndpoint is enabled, add the serviceAccountToken to QP via a projected volume
 	if cfg.Deployment.ConcurrencyStateEndpoint != "" {
+		// add token for audience "concurrency-state-hook" under filename ConcurrencyStateTokenFilename
+		addToken(tokenVolume, queue.ConcurrencyStateTokenFilename, concurrencyStateHook, ptr.Int64(600))
+	}
+
+	audiances := make([]string, 0, len(cfg.Deployment.QueueSidecarTokenAudiences))
+	for k := range cfg.Deployment.QueueSidecarTokenAudiences {
+		audiances = append(audiances, k)
+	}
+	sort.Strings(audiances)
+	for _, aud := range audiances {
+		// add token for audience <aud> under filename <aud>
+		addToken(tokenVolume, aud, aud, ptr.Int64(3600))
+	}
```

Comment:
```suggestion
	audiences := make([]string, 0, len(cfg.Deployment.QueueSidecarTokenAudiences))
	for k := range cfg.Deployment.QueueSidecarTokenAudiences {
		audiences = append(audiences, k)
	}
	sort.Strings(audiences)
	for _, aud := range audiences {
		// add token for audience <aud> under filename <aud>
		addToken(tokenVolume, aud, aud, ptr.Int64(3600))
	}
```

You can also write this as:

```suggestion
	for _, aud := range cfg.Deployment.QueueSidecarTokenAudiences.List() {
		// add token for audience <aud> under filename <aud>
		addToken(tokenVolume, aud, aud, ptr.Int64(3600))
	}
```

Because `List` returns a sorted list, per documentation: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String.List

Reply:
This is not readable  - I would expect a function called "List()" to sort the list as well - I would expect it to be called SortedList() :) 
Lets skip this specific change.

Reply:
`sets.String` also has an `UnsortedList`; presumably, they wanted `List` to work for people without the controller reordering you were worried about.

I'm going to approve, but since this is from the `k8s.io/apimachinery/pkg/util` library, I think it was written with the k8s concern in mind.